votepy/rules/greedy_monroe.py

```diff
@@ -0,0 +1,86 @@
+from votepy.ordinal_election import OrdinalElection, OrdinalBallot
+from typing import Union
+
+from votepy.structure.structure import algo, BaseAlgorithm, rule, impl
+from votepy.solve import solve
+
+
+@algo(name='greedy_monroe')
+class GreedyMonroe(BaseAlgorithm):
+    def _solve(self, voting: Union[OrdinalElection, list[int]], size_of_committee: int) -> list[int]:
+        m_candidates = voting.ballot_size
+        n_votes = len(voting)
+        if size_of_committee > m_candidates or size_of_committee <= 0:
+            raise ValueError(f"Size of committee needs to be from the range 1 to the number of all candidates.")
+        resultant_committee = []
+
+        def map_votes(vs):
+            mapping = {p: {} for p in range(n_votes)}
+            for i in range(n_votes):
+                for j in range(m_candidates):
+                    mapping[i][vs[i][j]] = m_candidates - j - 1
+            return mapping
+
+        mapping = map_votes(voting)
+        kk = size_of_committee
+        curr_candidates = set(range(m_candidates))
+
+        n = n_votes
+
+        for i in range(size_of_committee):
+            curr_size = n // kk
+            best_score, best_candidate = -1, -1
+            best_votes, best_candidate_voters = None, None
+            for candidate in curr_candidates:
+                calculate_scoring = [(j, mapping[j][candidate]) for j in range(n_votes) if mapping[j] is not None]
```

Comment:
`calculate_scoring` -> `calculated_scoring` or just `scoring`

Reply:
Maybe move this list comprehension to separate helper function? (like `map_votes`)

It is hard to read right now